src/gt4py/next/utils.py

```diff
@@ -285,213 +282,100 @@ def equalize_tuple_structure(
     return d1, d2
 
 
-class CanonicalizationOptions(NamedTuple):
-    function_name: str
-    allow_kwargs_mutation: bool
-    sort_kwargs: bool
-
-
-class CallArgsCanonicalizer(Protocol):
-    def __call__(self, args: tuple, kwargs: dict[str, Any]) -> tuple[tuple, dict[str, Any]]: ...
-
-    @property
-    def options(self) -> CanonicalizationOptions: ...
-
-    def cache_info(self) -> functools._CacheInfo: ...
-    def cache_clear(self) -> None: ...
-
-
-class CustomCallArgsCanonicalizerFactory(Protocol):
-    def __call__(
-        self, passed_pos_args_count: int, passed_kwargs_keys: tuple[str, ...]
-    ) -> CallArgsCanonicalizer: ...
-
-
-def make_args_canonicalizer_factory(
+def make_args_canonicalizer(
     signature: inspect.Signature,
     *,
-    name: str = "unknown",
-    allow_kwargs_mutation: bool = True,
-    sort_kwargs: bool = False,
-) -> CustomCallArgsCanonicalizerFactory:
+    name: str | None = None,
+) -> Callable[..., tuple[tuple, dict[str, Any]]]:
     """
-    Create a factory for functions that canonicalize call arguments for a given signature.
+    Create a call arguments canonicalizer function from a given signature.
 
-    For a full description of what canonicalization means, see `make_args_canonicalizer`.
+    The canonicalization means that the returned arguments are as all positional
```

Comment:
```suggestion
    The canonicalization means that the returned arguments are as if all positional
```
is that what you meant?

Reply:
The whole paragraph has been rephrased.